python/ray/experimental/workflow/storage/debug.py

```diff
@@ -0,0 +1,153 @@
+import json
+from typing import Any, List
+from urllib import parse
+import pathlib
+from filelock import FileLock
+from ray.experimental.workflow.storage import DEBUG_PREFIX
+from ray.experimental.workflow.storage.base import Storage
+from ray.experimental.workflow.storage.filesystem import FilesystemStorageImpl
+import ray.cloudpickle
+from ray.experimental.workflow import serialization_context
+
+
+class LoggedStorage(FilesystemStorageImpl):
+    """A storage records all writing to storage sequentially."""
+
+    def __init__(self, workflow_root_dir: str):
+        super().__init__(workflow_root_dir)
+        self._log_dir = self._workflow_root_dir / "log"
+        self._count = self._workflow_root_dir / "count.log"
+        if not self._log_dir.exists():
+            self._log_dir.mkdir()
+        # only one process initializes the count
+        with FileLock(str(self._workflow_root_dir / ".lock")):
+            if not self._count.exists():
+                with open(self._count, "x") as f:
+                    f.write("0")
+
+    async def put(self, key: str, data: Any, is_json: bool = False) -> None:
+        with FileLock(str(self._workflow_root_dir / ".lock")):
+            with open(self._count, "r") as f:
+                count = int(f.read())
+            k1 = self._log_dir / f"{count}.metadata.json"
+            k2 = self._log_dir / f"{count}.value"
+            await super().put(
```

Comment:
should we just store it in this db's dir?

Reply:
make sense